src/governance-vote.h

```diff
@@ -68,6 +68,8 @@ enum vote_signal_enum_t  {
     VOTE_SIGNAL_CUSTOM20   = 35
 };
 
+static const int MAX_SUPPORTED_VOTE_SIGNAL = 4;
```

Comment:
>That would require changing the signal enum

yep, very simple change, see the code I posted above

> as well as the supporting functions (ie. string to enum, enum to string)

Why? The shouldn't cause any issues for conversion, no need to `switch` by `max_...` or produce another string for it.

>I'm very skeptical of that particular "feature" 

That is actually very handy one, not sure why you are skeptical... This way if you need to change underlying values all "dependencies" will be resolved automatically. Say you need to change `VOTE_SIGNAL_ENDORSED` to `42` for some reason - having the code I posted above everything would still be consistent, having separate constant with magic number somewhere outside of enum would break things and you would have to find and modify it too.